### PR TITLE
Fix various warnings when building on GCC 5.2.1 on i386 (Ubuntu 15.10)

### LINKED ISSUE #1866: [1.3.0] Can't compile 32-bit overlay library
Using the latest Git, I am unable to compile a 32-bit overlay library that compliments the 64-bit one. It fails with this error:

 gcc -c -m32 -O2 -march=i486 -mtune=i686 -fvisibility=hidden -Wall -Wextra -Werror  -O2 -Wall -W -fPIC -DRESTRICT=__restrict__ -D_FORTIFY_SOURCE=2 -DTARGET_UNIX -I/usr/lib64/qt/mkspecs/linux-g++ -I. -o release/overlay.o overlay.c
overlay.c: In function 'drawOverlay':
overlay.c:348:5: error: format '%ld' expects argument of type 'long int', but argument 3 has type 'ssize_t' [-Werror=format=]
     ods("Short overlay message read %x %ld/%d", ctx->omMsg.omh.uiType, length, ctx->omMsg.omh.iLength);
     ^
overlay.c:366:25: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
         if (buf.st_size >= ctx->uiWidth * ctx->uiHeight * 4
                         ^
cc1: all warnings being treated as errors
make[1]: *** [release/overlay.o] Error 1
make[1]: Leaving directory `/tmp/SBo/mumble-1.3.0/overlay_gl'
make: *** [release] Error 2

Any ideas?

### PR DESCRIPTION
**Pull request by mkrautz**
Fixes #1866

### CODE REVIEW

File: overlay_gl/overlay.c [hunk 1]
```diff
@@ -345,7 +345,7 @@ static void drawOverlay(Context *ctx, unsigned int width, unsigned int height) {
 				disconnect(ctx);
 				return;
 			} else if (length != ctx->omMsg.omh.iLength) {
-				ods("Short overlay message read %x %ld/%d", ctx->omMsg.omh.uiType, length, ctx->omMsg.omh.iLength);
+				ods("Short overlay message read %x %zd/%d", ctx->omMsg.omh.uiType, length, ctx->omMsg.omh.iLength);
```
Comment (hacst) [inline overlay_gl/overlay.c]: Fine here. We have to make sure not to accidentally use "z" on Windows though as it doesn't support those even with VC14 (which is a pita).

**Comment by hacst** [comment]
LGTM

**Comment by Xylemon** [comment]
I can confirmed this fixed the issue for me!

**Comment by mkrautz** [comment]
Landed as
f32343d59770cb154fea1e19e2cc9cf0255bdbde
b72daa87db6e27393ea044152bf0d46cd7
e4c80dac75bd45f4fb18ff4a97a47a939d275919

**Pull request closed without merging**